### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by Maxwell011**
I have implemented 👍 

- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for editing task descriptions.

Kindly Review 🙏

### CODE REVIEW

File: modules/functions.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+// function for the clear completed button, hint: its not functional yet
+const clearCompleted = (index) => {
+  const checkedTask = document.querySelectorAll('.checkedTask');
+  checkedTask.forEach((task) => {
+    if (task.completed === true) {
+      removeTask(index);
+    }
+  });
+};
```
Comment (GzimAsani) [inline modules/functions.js:119]: - [ ] Clear the completed tasks are not working, kindly look into that https://www.loom.com/share/a11bf2ccb1e647438eb67e37b7d01dee

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @Maxwell011    :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your project, adding removing and saving to local storage works  :+1:  
 
 ## Required Changes 

- [ ] Check the comment below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/functions.js (hunk 1 above)
Comment (Maxwell011) [reply modules/functions.js]: I have fixed the changes if you Add, then click the checkbox and clicked clear completed it deletes

**Review by Ugboaja-Uchechi — CHANGES_REQUESTED** [review]
Hi @Maxwell011,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- Linters are passing ✔️ 
- README is descriptive 👍 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**